luda-adventure/src/app/game/update/resolve_collision_about_character.rs

```diff
@@ -0,0 +1,129 @@
+use crate::{
+    app::game::{
+        known_id::object::PLAYER_CHARACTER, Collider, CollisionBox, Game, PlayerCharacter,
+        Positioner, Tile, TileExt,
+    },
+    ecs,
+};
+use namui::prelude::*;
+
+impl Game {
+    pub fn resolve_collision_about_character(&mut self) {
+        let collision_box_list_except_character =
+            get_collision_box_list_except_character(&self.ecs_app);
+        if let Some((_, (_, character_collider, character_positioner))) = self
+            .ecs_app
+            .query_entities_mut::<(&PlayerCharacter, &Collider, &mut Positioner)>()
+            .first_mut()
+        {
+            for other_collision_box in collision_box_list_except_character {
+                let character_collision_box =
+                    character_collider.get_collision_box(character_positioner.xy());
+                if !detect_collision(character_collision_box, other_collision_box) {
+                    continue;
+                }
+
+                let (collision_normal, penetration_depth) =
+                    get_collision_normal_and_penetration_depth(
+                        character_collision_box,
+                        other_collision_box,
+                    );
+                move_back_by_penetration_depth(
+                    collision_normal,
+                    penetration_depth,
+                    character_positioner,
+                );
+            }
+        }
+    }
+}
```

Comment:
doesn't it keep character seems to move to wall repeatedly even if it collide to wall but by interpolation?

Reply:
No, it won't. due to `move_back_by_penetration_depth`.
In end of tick, positioner has collision resolved xy. and `save_xy_for_interpolation` always saves collision resolved xy.
![Animation](https://user-images.githubusercontent.com/38313680/197386413-48322699-883c-44ad-9dcc-6128284e5e0a.gif)